apps/cowswap-frontend/src/modules/swap/hooks/useCheckEthFlowOrderExists.ts

```diff
@@ -0,0 +1,39 @@
+import { useAtomValue } from 'jotai'
+import { useCallback } from 'react'
+
+import { ZERO_ADDRESS } from 'legacy/constants/misc'
+import { useEthFlowContract } from 'legacy/hooks/useContract'
+
+import { ethFlowInFlightOrderIdsAtom } from '../state/EthFlow/ethFlowInFlightOrderIdsAtom'
+
+export interface EthFlowOrderExistsCallback {
+  (orderId: string, orderDigest: string): Promise<boolean>
+}
+
+export function useCheckEthFlowOrderExists(): EthFlowOrderExistsCallback {
+  const ethFlowInFlightOrderIds = useAtomValue(ethFlowInFlightOrderIdsAtom)
+  const ethFlowContract = useEthFlowContract()
+
+  return useCallback(
+    async (orderId: string, orderDigest: string) => {
+      if (ethFlowInFlightOrderIds.includes(orderId)) {
+        return true
+      }
+
+      if (ethFlowContract) {
+        try {
+          const { owner } = await ethFlowContract.callStatic.orders(orderDigest)
+
+          return owner.toLowerCase() !== ZERO_ADDRESS
+        } catch (e) {
+          console.error('Eth-flow order existing check error: ', e)
+
+          return false
```

Comment:
So if we have an error, we take a bet :) 

okok, i guess is better than not continuing with the flow, and re-attempts might be an overkill.